curvefs/src/client/client_operator.cpp

```diff
@@ -209,22 +209,64 @@ CURVEFS_ERROR RenameOperator::CommitTx() {
     return CURVEFS_ERROR::OK;
 }
 
-void RenameOperator::UnlinkOldInode() {
-    if (oldInodeId_ == 0) {
-        return;
+CURVEFS_ERROR RenameOperator::LinkInode(uint64_t inodeId) {
+    std::shared_ptr<InodeWrapper> inodeWrapper;
+    auto rc = inodeManager_->GetInode(inodeId, inodeWrapper);
+    if (rc != CURVEFS_ERROR::OK) {
+        LOG_ERROR("GetInode", rc);
+        return rc;
     }
 
+    rc = inodeWrapper->LinkLocked();
+    if (rc != CURVEFS_ERROR::OK) {
+        LOG_ERROR("Link", rc);
+        return rc;
+    }
+
+    // CURVEFS_ERROR::OK
+    return rc;
+}
+
+CURVEFS_ERROR RenameOperator::UnLinkInode(uint64_t inodeId) {
     std::shared_ptr<InodeWrapper> inodeWrapper;
-    auto rc = inodeManager_->GetInode(oldInodeId_, inodeWrapper);
+    auto rc = inodeManager_->GetInode(inodeId, inodeWrapper);
     if (rc != CURVEFS_ERROR::OK) {
         LOG_ERROR("GetInode", rc);
-        return;
+        return rc;
     }
 
     rc = inodeWrapper->UnLinkLocked();
     if (rc != CURVEFS_ERROR::OK) {
         LOG_ERROR("UnLink", rc);
+        return rc;
     }
+
+    // CURVEFS_ERROR::OK
+    return rc;
+}
+
+CURVEFS_ERROR RenameOperator::LinkDestParentInode() {
+    if (parentId_ == newParentId_) {  // under same directory
+        return CURVEFS_ERROR::OK;
+    }
+    return LinkInode(newParentId_);
+}
+
+void RenameOperator::UnlinkSrcParentInode() {
+    if (parentId_ == newParentId_) {  // under same directory
+        return;
+    }
+    auto rc = UnLinkInode(parentId_);
+    LOG(INFO) << "Unlink source parent inode, retCode = " << rc;
+}
+
+void RenameOperator::UnlinkOldInode() {
+    if (oldInodeId_ == 0) {
```

Comment:
if threre is old inode here,  UnlinkSrcParentInode should not be called.

Reply:
> if threre is an old inode here, UnlinkSrcParentInode should not be called?

The `oldInodeId_` means the inode of removed dentry, not its directory.

Reply:
sorry, LinkDestParentInode should not be called

Reply:
> sorry, LinkDestParentInode should not be called

fixed.